src/basic/time-util.c

```diff
@@ -1604,15 +1605,73 @@ bool clock_supported(clockid_t clock) {
         }
 }
 
+int find_localtime_last_symlink(char **ret) {
+        _cleanup_free_ char *path = NULL, *last_symlink = NULL;
+        int r;
+        unsigned i;
+
+        assert(ret);
+
+        path = strdup("/etc/localtime");
+        if (!path)
+                return log_oom();
```

Comment:
this function doesn't log, so just return -ENOMEM here and below, similar for other cases